sources/packages/backend/apps/workers/src/services/student-assessment/student-assessment.service.ts

```diff
@@ -170,12 +174,81 @@ export class StudentAssessmentService extends RecordDataModelService<StudentAsse
     assessmentId: number,
     status: AssessmentStatus,
   ): Promise<UpdateResult> {
-    return this.repo.update(
+    return await this.dataSource.transaction(
+      async (transactionalEntityManager) => {
+        const updateResult = await transactionalEntityManager
+          .getRepository(StudentAssessment)
+          .createQueryBuilder()
+          .update(StudentAssessment)
+          .set({
+            noaApprovalStatus: status,
+          })
+          .where("id = :assessmentId", { assessmentId })
```

Comment:
Please refrain from creating string-based queries unless necessary.

Reply:
actually, it was already there, I just updated it to a transaction